packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -211,10 +221,30 @@ export class TokenDetectionController extends BaseController<
     }, this.config.interval);
   }
 
+  private getCorrectChainId(networkClientId?: NetworkClientId) {
+    if (networkClientId) {
+      return this.getNetworkClientById(networkClientId).configuration.chainId;
+    }
+    return this.config.chainId;
+  }
+
+  _executePoll(
+    networkClientId: string,
+    options: { address: string },
+  ): Promise<void> {
+    return this.detectTokens(networkClientId, options?.address);
```

Comment:
It appears based on the type that `options` will always be there. Do we need the `?.`?

``` suggestion
    return this.detectTokens(networkClientId, options?.address);